app/controllers/hosts_controller.rb

```diff
@@ -454,6 +454,22 @@ def update_multiple_power_state
     redirect_back_or_to hosts_path
   end
 
+  def show_multiple_power_status
+    res = {}
+    if @hosts.respond_to? :map
+      res = @hosts.map do |host|
+        begin
+          { host.id => host.supports_power_and_running? }
+        rescue Foreman::Exception
+          {}
+        end
+      end.reduce(:merge)
+    else
+      logger.warn "Malformed host_ids parameter - not and array"
```

Comment:
It seems we can get here: ![](https://github.com/theforeman/foreman/pull/3459#issuecomment-224290513)